plugins/window-state/src/lib.rs

```diff
@@ -278,10 +274,11 @@ impl<R: Runtime> WindowExtInternal for WebviewWindow<R> {
 
 impl<R: Runtime> WindowExtInternal for Window<R> {
     fn update_state(&self, state: &mut WindowState, flags: StateFlags) -> tauri::Result<()> {
-        let is_maximized = match flags.intersects(StateFlags::MAXIMIZED | StateFlags::SIZE) {
-            true => self.is_maximized()?,
-            false => false,
-        };
+        let is_maximized = flags
+            .intersects(StateFlags::MAXIMIZED | StateFlags::POSITION | StateFlags::SIZE)
+            && self.is_maximized()?;
+        let is_minimized =
+            flags.intersects(StateFlags::POSITION | StateFlags::SIZE) && self.is_minimized()?;
```

Comment:
why are we checking for exitence of `Postiion` and `Size` flags here? and later check them again. doesn't make sense, it should be checked later only

Reply:
Because we use this info on those places only and this is so we can re-use the `is_maximized` and `is_minimized` result

Reply:
I am only talking about flags `flags
            .intersects(StateFlags::POSITION | StateFlags::SIZE)`

Reply:
ah nvm, I am dumb